tensorflow/python/ops/nn_ops.py

```diff
@@ -4193,7 +4194,7 @@ def _get_noise_shape(x, noise_shape):
 @deprecation.deprecated_args(None, "Please use `rate` instead of `keep_prob`. "
                              "Rate should be set to `rate = 1 - keep_prob`.",
                              "keep_prob")
-def dropout(x, keep_prob=None, noise_shape=None, seed=None, name=None,
+def dropout(x, keep_prob=None, noise_shape=None, seed=0, name=None,
```

Comment:
Note, this will change the current dropout interface. 

If seed is a input, `None` value cannot be converted to a tensor.

If seed is a attribute, the only available type is int, which is a int32. However, a python integer can be larger than int32.

Based on above, seed can only be a input to cope with original value ranges. This means that the default value of seed needs to update to 0 instead of None.

Reply:
Reverted this interface change. This is causing a problem in a number of non-drop related edge cases